### ISSUE TITLE
simpleAI example problem

### ISSUE THREAD

**Issue by PetterSunnyVR** [meniku/NPBehave#34]
```
new Sequence(

                            // Do once 
                            new NBAction(() => {
                                                
                                                SetColor(Color.red); 
                                                }) 
                            { Label = "Change to Red" },

                            // go towards player as long as playerDistance is less than 7.5 ( in that case, _shouldCancel is false )
                            new NBAction((bool _shouldCancel) =>
                            {
                                //_shouldCancel turns TRUE when playerDiscance is more than 7.5 => means " do I have to abort"
                                if (!_shouldCancel)
                                {
                                    MoveTowards(blackboard.Get<Vector3>("playerLocalPos"));
                                    //inform that this action is being performed until the condition changes => waits for distance to be greater (it will run else statement)
                                    return NBAction.Result.PROGRESS;
                                }
                                else
                                {
                                    
                                    //upon returning false starts the next Node in the Node[]
                                    return NBAction.Result.SUCCESS;
                                }
                            })
                            { Label = "Follow" }
                            ,
                            new NBAction(() => {
                                Debug.Log("yellow");
                                SetColor(Color.yellow);
                            })
                            { Label = "Change to Yellow" }
                            
                        );
```

It is from simpleAI example. It is run inside a condition with Stops.IMMEDIATE_RESTART flag.
I don't quite understand why when I return SUCCESS in the second action it never runs the Change to Yellow action. If I change PROGRESS to SUCCESS it runs all 3 Actions constantly. Why doesn't it run the last action before restarting if we stop returning PROGRESS and return SUCCESS instead? I thought Sequence should finish all its actions if we don't return FAILED ? 
In other words how would I add another condition before we restart ?

**Comment by meniku** [comment]
I think you hit a weekness in the way abortion [currently works](https://github.com/meniku/NPBehave/issues/23#issuecomment-541451569).

For [2.0](https://github.com/meniku/NPBehave/issues/25) I'm planning to rewrite the aborting to get rid of that weakness

To solve your issue, what I would suggest you is to remove the "RED" and "YELLOW" and wrap the whole `Sequence` inside an `Observer` decorator.
The observer decorator has a `onStart` and `onStop` Action where. You can implement the "RED" logic in `onStart` and the "YELLOW" logic in the `onStop`, this should guarantee that the you'll end up YELLOW in whatever case the branch is left.

I hope that works for you

**Comment by PetterSunnyVR** [comment]
Thanks a lot for the info. I have switched to Panda BT as it is much easier to work with. I really like this implementation though.

**Comment by meniku** [comment]
fair enough. I'll close this issue then.
